commit cb74c432e321ed645b6cd88b77edc15f9478efbd
Author: Joseph Jezak <josejx@gentoo.org>
Date:   Sun Jun 11 12:00:37 2006 -0400

    [PATCH] SoftMAC: Prevent multiple authentication attempts on the same network
    
    This patch addresses the "No queue exists" messages commonly seen during
    authentication and associating.  These appear due to scheduling multiple
    authentication attempts on the same network.  To prevent this, I added a
    flag to stop multiple authentication attempts by the association layer.
    I also added a check to the wx handler to see if we're connecting to a
    different network than the one already in progress.  This scenario was
    causing multiple requests on the same network because the network BSSID
    was not being updated despite the fact that the ESSID changed.
    
    Signed-off-by: Joseph Jezak <josejx@gentoo.org>
    Signed-off-by: John W. Linville <linville@tuxdriver.com>

diff --git a/include/net/ieee80211softmac.h b/include/net/ieee80211softmac.h
index 7a483ab4022f..00ad810eb883 100644
--- a/include/net/ieee80211softmac.h
+++ b/include/net/ieee80211softmac.h
@@ -104,6 +104,7 @@ struct ieee80211softmac_assoc_info {
 	 */
 	u8 static_essid:1,
 	   associating:1,
+	   assoc_wait:1,
 	   bssvalid:1,
 	   bssfixed:1;